Adopt more spans in VectorMath.h

cdumez opened the pull request:
#### 55186327a6c41b86f59028fcffcd41f49dd4e56c
<pre>
Adopt more spans in VectorMath.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=285228">https://bugs.webkit.org/show_bug.cgi?id=285228</a>

Reviewed by Darin Adler.

* Source/WebCore/Modules/webaudio/AudioParam.cpp:
(WebCore::AudioParam::calculateFinalValues):
* Source/WebCore/Modules/webaudio/AudioParamTimeline.cpp:
(WebCore::AudioParamTimeline::valuesForFrameRange):
(WebCore::AudioParamTimeline::processLinearRamp):
(WebCore::AudioParamTimeline::processSetTarget):
* Source/WebCore/Modules/webaudio/DelayDSPKernel.cpp:
(WebCore::DelayDSPKernel::processKRate):
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
(WebCore::OscillatorNode::calculateSampleAccuratePhaseIncrements):
* Source/WebCore/Modules/webaudio/PeriodicWave.cpp:
(WebCore::PeriodicWave::createBandLimitedTables):
* Source/WebCore/Modules/webaudio/RealtimeAnalyser.cpp:
(WebCore::RealtimeAnalyser::getFloatFrequencyData):
* Source/WebCore/platform/audio/AudioBus.cpp:
(WebCore::AudioBus::speakersSumFromByDownMixing):
(WebCore::AudioBus::copyWithGainFrom):
(WebCore::AudioBus::copyWithSampleAccurateGainValuesFrom):
(WebCore::AudioBus::createByMixingToMono):
* Source/WebCore/platform/audio/AudioChannel.cpp:
(WebCore::AudioChannel::scale):
(WebCore::AudioChannel::sumFrom):
(WebCore::AudioChannel::maxAbsValue const):
* Source/WebCore/platform/audio/EqualPowerPanner.cpp:
(WebCore::EqualPowerPanner::pan):
* Source/WebCore/platform/audio/FFTConvolver.cpp:
(WebCore::FFTConvolver::process):
* Source/WebCore/platform/audio/FFTFrame.cpp:
(WebCore::FFTFrame::scaleFFT):
(WebCore::FFTFrame::multiply):
* Source/WebCore/platform/audio/IIRFilter.cpp:
(WebCore::IIRFilter::tailTime):
* Source/WebCore/platform/audio/Reverb.cpp:
(WebCore::calculateNormalizationScale):
* Source/WebCore/platform/audio/ReverbAccumulationBuffer.cpp:
(WebCore::ReverbAccumulationBuffer::accumulate):
* Source/WebCore/platform/audio/ReverbConvolverStage.cpp:
(WebCore::ReverbConvolverStage::ReverbConvolverStage):
* Source/WebCore/platform/audio/StereoPanner.cpp:
(WebCore::StereoPanner::panToTargetValue):
* Source/WebCore/platform/audio/VectorMath.cpp:
(WebCore::VectorMath::multiplyByScalar):
(WebCore::VectorMath::substract):
(WebCore::VectorMath::addScalar):
(WebCore::VectorMath::multiply):
(WebCore::VectorMath::interpolate):
(WebCore::VectorMath::multiplyComplex):
(WebCore::VectorMath::multiplyByScalarThenAddToOutput):
(WebCore::VectorMath::multiplyByScalarThenAddToVector):
(WebCore::VectorMath::addVectorsThenMultiplyByScalar):
(WebCore::VectorMath::maximumMagnitude):
(WebCore::VectorMath::sumOfSquares):
(WebCore::VectorMath::clamp):
(WebCore::VectorMath::linearToDecibels):
(WebCore::VectorMath::add):
* Source/WebCore/platform/audio/VectorMath.h:
* Source/WebCore/platform/audio/mac/FFTFrameMac.cpp:
(WebCore::FFTFrame::doFFT):
(WebCore::FFTFrame::doInverseFFT):

Canonical link: <a href="https://commits.webkit.org/288334@main">https://commits.webkit.org/288334@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/89f25733b5f2482262c90498fe6088b53ab5ea13

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82589 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2242 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36720 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/87711 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33647 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84685 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2314 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10148 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64365 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22123 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85648 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1673 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75132 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44640 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1569 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29318 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/32681 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72857 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29972 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/89075 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9884 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7145 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72773 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10112 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70942 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71987 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16129 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/15102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1273 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12809 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9837 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15358 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9711 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13176 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11480 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/audio/IIRFilter.cpp:
```diff
@@ -201,16 +201,18 @@ double IIRFilter::tailTime(double sampleRate, bool isFilterStable)
-    magnitudes[0] = VectorMath::maximumMagnitude(output.data(), AudioUtilities::renderQuantumSize);
+    auto inputSpan = input.span().first(AudioUtilities::renderQuantumSize);
+    auto outputSpan = output.span().first(AudioUtilities::renderQuantumSize);
+    process(inputSpan, output.span());
```
Why is this using `output.span()` instead of `outputSpan`?

cdumez replied:
My bad, will fix.

darinadler reviewed (approved):
Seems like some of the calls to `first` aren't needed but probably not worth more study to figure out which ones. Or maybe you did that carefully already.

cdumez committed:
Adopt more spans in VectorMath.h
https://bugs.webkit.org/show_bug.cgi?id=285228

Reviewed by Darin Adler.

* Source/WebCore/Modules/webaudio/AudioParam.cpp:
(WebCore::AudioParam::calculateFinalValues):
* Source/WebCore/Modules/webaudio/AudioParamTimeline.cpp:
(WebCore::AudioParamTimeline::valuesForFrameRange):
(WebCore::AudioParamTimeline::processLinearRamp):
(WebCore::AudioParamTimeline::processSetTarget):
* Source/WebCore/Modules/webaudio/DelayDSPKernel.cpp:
(WebCore::DelayDSPKernel::processKRate):
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
(WebCore::OscillatorNode::calculateSampleAccuratePhaseIncrements):
* Source/WebCore/Modules/webaudio/PeriodicWave.cpp:
(WebCore::PeriodicWave::createBandLimitedTables):
* Source/WebCore/Modules/webaudio/RealtimeAnalyser.cpp:
(WebCore::RealtimeAnalyser::getFloatFrequencyData):
* Source/WebCore/platform/audio/AudioBus.cpp:
(WebCore::AudioBus::speakersSumFromByDownMixing):
(WebCore::AudioBus::copyWithGainFrom):
(WebCore::AudioBus::copyWithSampleAccurateGainValuesFrom):
(WebCore::AudioBus::createByMixingToMono):
* Source/WebCore/platform/audio/AudioChannel.cpp:
(WebCore::AudioChannel::scale):
(WebCore::AudioChannel::sumFrom):
(WebCore::AudioChannel::maxAbsValue const):
* Source/WebCore/platform/audio/EqualPowerPanner.cpp:
(WebCore::EqualPowerPanner::pan):
* Source/WebCore/platform/audio/FFTConvolver.cpp:
(WebCore::FFTConvolver::process):
* Source/WebCore/platform/audio/FFTFrame.cpp:
(WebCore::FFTFrame::scaleFFT):
(WebCore::FFTFrame::multiply):
* Source/WebCore/platform/audio/IIRFilter.cpp:
(WebCore::IIRFilter::tailTime):
* Source/WebCore/platform/audio/Reverb.cpp:
(WebCore::calculateNormalizationScale):
* Source/WebCore/platform/audio/ReverbAccumulationBuffer.cpp:
(WebCore::ReverbAccumulationBuffer::accumulate):
* Source/WebCore/platform/audio/ReverbConvolverStage.cpp:
(WebCore::ReverbConvolverStage::ReverbConvolverStage):
* Source/WebCore/platform/audio/StereoPanner.cpp:
(WebCore::StereoPanner::panToTargetValue):
* Source/WebCore/platform/audio/VectorMath.cpp:
(WebCore::VectorMath::multiplyByScalar):
(WebCore::VectorMath::substract):
(WebCore::VectorMath::addScalar):
(WebCore::VectorMath::multiply):
(WebCore::VectorMath::interpolate):
(WebCore::VectorMath::multiplyComplex):
(WebCore::VectorMath::multiplyByScalarThenAddToOutput):
(WebCore::VectorMath::multiplyByScalarThenAddToVector):
(WebCore::VectorMath::addVectorsThenMultiplyByScalar):
(WebCore::VectorMath::maximumMagnitude):
(WebCore::VectorMath::sumOfSquares):
(WebCore::VectorMath::clamp):
(WebCore::VectorMath::linearToDecibels):
(WebCore::VectorMath::add):
* Source/WebCore/platform/audio/VectorMath.h:
* Source/WebCore/platform/audio/mac/FFTFrameMac.cpp:
(WebCore::FFTFrame::doFFT):
(WebCore::FFTFrame::doInverseFFT):

Canonical link: https://commits.webkit.org/288334@main
